subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/DefaultGenericFileTreeSnapshotter.java

```diff
@@ -55,7 +56,8 @@ public void visitFile(FileVisitDetails fileDetails) {
                     fileDetails.getName(),
                     DefaultFileMetadataSnapshot.file(
                         fileDetails.getLastModified(),
-                        fileDetails.getSize()
+                        fileDetails.getSize(),
+                        AccessType.DIRECT
```

Comment:
I like the use of an enum. 👍

Can we leave a comment here about why this is assumed direct?

Reply:
It is not assumed `DIRECT`. This is the current code:
```
                    DefaultFileMetadataSnapshot.file(
                        fileDetails.getLastModified(),
                        fileDetails.getSize(),
                        Files.isSymbolicLink(Paths.get(fileDetails.getPath())) ? AccessType.VIA_SYMLINK : AccessType.DIRECT
                    )
```

Did you see the last commit?